repair/repair.cc

```diff
@@ -936,6 +936,11 @@ struct repair_options {
     }
 };
 
+static future<tasks::task_manager::task_ptr> start_repair_task(tasks::task_manager::task::task_impl_ptr task_impl_ptr, shared_ptr<repair_module> module, tasks::task_info pd = {}) {
+    auto task = co_await module->make_task(std::move(task_impl_ptr), pd);
+    task->start();
```

Comment:
start should handle the aborted case where the task is already aborted at this point (if the parent task is already aborted), so it doesn't set the task state to `running`.
We should probably run the "monitor" fiber that calls `unregister_task` after the ttl expires,
but not call `run_to_completion`.

Reply:
Do we really need to check parent task? I think that `this->abort_requested()` would be enough:
- if parent is aborted, then task is created -> task is aborted
- if task is created, then parent is aborted, then task is started -> task abortion is populated (and maybe done).

Also, then `start()` would be able to still return `void` and not `future<>`, which may be useful in some cases.

Reply:
I didn't see we need to check the parent task :)
You're right, if the parent task is already aborted, make_task will return a task that's already aborted as well.
I'm saying that task->start() should handle this case and e.g. refrain from calling `run_to_completion()`.

Reply:
Done: #11995